crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -117,29 +164,33 @@ impl Inner {
         max_age: Duration,
     ) -> futures::stream::BoxStream<'a, (H160, NativePriceEstimateResult)> {
         let estimates = tokens.iter().map(move |token| async move {
-            {
-                // check if price is cached by now
+            let current_accumulative_errors_count = {
+                // check if the price is cached by now
                 let now = Instant::now();
                 let mut cache = self.cache.lock().unwrap();
-                let price = Self::get_cached_price(*token, now, &mut cache, &max_age, false);
-                if let Some(price) = price {
-                    return (*token, price);
+
+                match Self::get_cached_price(*token, now, &mut cache, &max_age, false) {
+                    Some(cached) => {
+                        if cached.is_ready() {
+                            return (*token, cached.result);
+                        } else {
+                            cached.accumulative_errors_count
+                        }
+                    }
+                    None => Default::default(),
```

Comment:
IMO slightly easier to read.
```suggestion
                    Some(cached) if cached.is_ready() => return (*token, cached.result),
                    Some(cached) => cached.accumulative_errors_count,
                    None => 0,
```